PlayerController.cs

```diff
@@ -0,0 +1,321 @@
+
+    private void Awake()
+    {
+        if (transform.localScale.x > 0)
+        {
+            facingRight = true;
+        }
+        saveManager = FindObjectOfType<SaveManager>();
+        blackScreenAnimator = FindObjectOfType<Canvas>().GetComponentInChildren<Animator>();
+        if (!saveManager.GetSaveExisting()) return;
+        saveManager.Load();
+
+        currentHealth = stats.health;
+        currentShield = stats.shield;
+    }
+
+    private void Start()
+    {
+        extraJumps = extraJumpsValue;
+        rb2d = GetComponent<Rigidbody2D>();
+
+        currentHealth = playerMaxHealth;
+        hpBar.SetMaxHealth(playerMaxHealth);
+        stats.maxHealth = playerMaxHealth;
+    }
+
+    public Animator anim;
+
+    private void TakeDamage(int dmgAmount)
+    {
+        hpBar.TakeDamage(dmgAmount);
+        currentHealth = hpBar.GetHealth();
+
+        if (hpBar.HasShieldActive()) return;
+        oofSound.Play(0);
+    }
+
+    public void Heal(int healAmount)
+    {
+        currentHealth = hpBar.GetHealth();
+        hpBar.Heal(healAmount);
+        healSound.Play(0);
+    }
+
+    public void GameOver()
+    {
+        rb2d.mass = 10000000;
+        rb2d.gravityScale = 1000000;
+        SceneManager.LoadScene("GameOverScene");
+    }
+
+    public void SetMaxHP(int maxHP)
+    {
+        hpBar.SetMaxHealth(maxHP);
+    }
+
+    private void FixedUpdate()
+    {
+        currentHealth = hpBar.GetHealth();
+        isGrounded = Physics2D.OverlapCircle(groundCheck.position, checkRadius, whatIsGround);
+        if (isGrounded && rb2d.velocity.y < -30)
+        {
+            TakeDamage(-(int)Math.Round(rb2d.velocity.y) - 10);
+        }
+
+        //moveInput = Input.GetAxisRaw("Horizontal");
+
+        if (Input.GetKey(GameManager.GM.left))
+        {
+            moveInput = -1 * speed;
+        }
+        else if (Input.GetKey(GameManager.GM.right))
+        {
+            moveInput = 1 * speed;
+        }
+        else moveInput = 0;
+
+        rb2d.velocity = new Vector2(moveInput, rb2d.velocity.y);
+
+        if (gameOver) return;
+        if (isInSomeMenu) return;
+        if (!flip) return;
+        switch (facingRight)
+        {
+            case false when moveInput > 0:
+            case true when moveInput < 0:
+                Flip();
+                break;
+        }
+    }
+
+    public void OnCollisionEnter2D(Collision2D targetObj)
+    {
+        if(targetObj.gameObject.tag.Equals("Danger"))
+        {
+            TakeDamage(10);
+            isInDangerPlatform = true;
+        }
+
+        if (targetObj.gameObject.tag.Equals("Heal"))
+        {
+            if (currentHealth >= 100) return;
+            Heal(10);
+            isInHealPlatform = true;
+        }
+
+        // Debug.Log(rb2d.velocity.y);
+        // if (targetObj.gameObject.layer.Equals(8) && rb2d.velocity.y < -10)
+        // {
+        //     
+        //     TakeDamage(-(int)Math.Round(rb2d.velocity.y) * 2);
+        // }
```

Comment:
Remove this code. Commented code is a bad practice. But comments that indicate or describe the action is desirable.